packages/test/src/test-data-converter.ts

```diff
@@ -62,11 +70,120 @@ test('JsonPayloadConverter converts to object', async (t) => {
+  t.deepEqual(await converter.toData(instance), {
+    metadata: {
+      [METADATA_ENCODING_KEY]: encodingKeys.METADATA_ENCODING_PROTOBUF,
+      [METADATA_MESSAGE_TYPE_KEY]: u8('ProtoActivityInput'),
+    },
+    data: root.ProtoActivityInput.encode(instance).finish(),
+  });
+});
+
+test('ProtobufBinaryPayloadConverter converts to an instance', async (t) => {
+  const instance = root.ProtoActivityInput.create({ name: 'Proto', age: 1 });
+  const converter = new ProtobufBinaryPayloadConverter(root);
+  const testInstance = await converter.fromData((await converter.toData(instance))!);
+  // tests that both are instances of the same class with the same properties
+  t.deepEqual(testInstance, instance);
+});
+
+test('ProtobufBinaryPayloadConverter throws detailed errors', async (t) => {
+  const instance = root.ProtoActivityInput.create({ name: 'Proto', age: 1 });
+  const converter = new ProtobufBinaryPayloadConverter(root);
+
+  await t.throwsAsync(
+    async () =>
+      await converter.fromData({
+        metadata: { [METADATA_ENCODING_KEY]: encodingKeys.METADATA_ENCODING_PROTOBUF },
+        data: root.ProtoActivityInput.encode(instance).finish(),
+      }),
+    { instanceOf: ValueError, message: 'Got protobuf payload without metadata.messageType' }
+  );
+  await t.throwsAsync(
+    async () =>
+      await converter.fromData({
+        metadata: {
+          [METADATA_ENCODING_KEY]: encodingKeys.METADATA_ENCODING_PROTOBUF,
+          [METADATA_MESSAGE_TYPE_KEY]: u8('NonExistentMessageClass'),
+        },
+        data: root.ProtoActivityInput.encode(instance).finish(),
+      }),
+    {
+      instanceOf: DataConverterError,
+      message: 'Got a `NonExistentMessageClass` protobuf message but cannot find corresponding message class in `root`',
+    }
+  );
+});
+
+test('ProtobufJSONPayloadConverter converts from an instance to JSON', async (t) => {
+  const instance = root.foo.bar.ProtoActivityInput.create({ name: 'Proto', age: 1 });
+  const converter = new ProtobufJsonPayloadConverter(root);
+  t.deepEqual(await converter.toData(instance), {
+    metadata: {
+      [METADATA_ENCODING_KEY]: encodingKeys.METADATA_ENCODING_PROTOBUF_JSON,
+      [METADATA_MESSAGE_TYPE_KEY]: u8('foo.bar.ProtoActivityInput'),
+    },
+    data: u8(JSON.stringify(instance)),
+  });
+});
+
+test('ProtobufJSONPayloadConverter converts to an instance from JSON', async (t) => {
+  const instance = root.foo.bar.ProtoActivityInput.create({ name: 'Proto', age: 1 });
+  const converter = new ProtobufJsonPayloadConverter(root);
+  const testInstance = await converter.fromData((await converter.toData(instance))!);
+  // tests that both are instances of the same class with the same properties
+  t.deepEqual(testInstance, instance);
+});
+
+test('ProtobufJSONPayloadConverter converts binary', async (t) => {
+  // binary should be base64-encoded:
+  // https://developers.google.com/protocol-buffers/docs/proto3#json
```

Comment:
The lib we're using seems to have good tests for the types listed in the spec: https://github.com/googleapis/proto3-json-serializer-nodejs/tree/main/typescript/test/unit

Let me know if you'd like me to:
- write tests for more types
- do the compact testing against the other SDKs (I imagine they also follow the spec and thus work, but haven't tried yet)

Reply:
No need, the initial comment was hinting that we should verify that we support the standard protobuf JSON serialization.